src/k8s/vmBuilder.js

```diff
@@ -282,3 +289,43 @@ export const getDataVolumeTemplateSpec = (storage, dvSource) => {
     },
   };
 };
+
+export const assignBootOrderIndex = (vm, currDevBootOrder = -1) => {
+  let bootOrder = currDevBootOrder;
+  if (currDevBootOrder !== BOOT_ORDER_FIRST) {
+    bootOrder = getNextAvailableBootOrderIndex(vm);
+  }
+  return bootOrder;
+};
+
+const getNextAvailableBootOrderIndex = vm => {
+  const sortedBootableDevices = getBootableDevicesInOrder(vm);
+  const numBootableDevices = sortedBootableDevices.length;
+
+  // assigned indexes start at two as the first index is assigned directly by the user
+  return numBootableDevices > 0 ? sortedBootableDevices[numBootableDevices - 1].value.bootOrder + 1 : BOOT_ORDER_SECOND;
```

Comment:
can we return 0 instead of BOOT_ORDER_SECOND ? and do this when we use it : 

`getNextAvailableBootOrderIndex(vm) || BOOT_ORDER_SECOND`

this way, the function will contain less magic

Reply:
I think it makes more sense as it's currently written. The function is just expected to return the next available index, so the caller wouldn't expect to have to handle a special case like that when using it. I just reused the existing constant to avoid having a magic number and to increase readability. 

I tweaked the last line a small bit to make it a bit more understandable at first glance. Thoughts?

Reply:
ok, your call :) 

We have just one caller of the method and I would rather move the hacky logic higher up the chain. I was more focused on the method's name (`getNextAvailableBootOrderIndex`) which doesn't adhere to the reality. 
Nevertheless both are valid approaches.

Reply:
Understood. I think the name is valid, though. All the hacky logic does is keep the method from returning an index of 1, which isn't technically an available index since it's assigned explicitly. I think I'll leave it as-is for the time being. We can always change it later if it becomes problematic.

Reply:
ok